perf/lua/benchmark.lua

```diff
@@ -119,13 +123,37 @@ function M.argparse(arg, argtable, custom_help)
     return params
 end
 
+local function load_average()
+    local path = '/proc/loadavg'
+    local fh = assert(fio.open(path, {'O_RDONLY'}))
+    local loadavg_buf = fh:read(1024)
+    fh:close()
+    -- Format is '0.89 0.66 0.80 2/1576 1203510'.
+    local loadavg_re = '(%d+.%d+) (%d+.%d+) (%d+.%d+)'
+
+    return { loadavg_buf:gmatch(loadavg_re)() }
+end
+
+local function perf_context()
+    local ctx = {}
+    ctx.date = os.date('%Y-%m-%d %H:%M:%S')
```

Comment:
The format of the date isn't the same as for Google Benchmark:
```json
{
  "date": "2024-07-16T11:59:25+03:00",

```

vs
```json
{
  "date": "2024-07-16 11:51:59",
```

Reply:
Example of date is `"date": "2015/03/17-18:40:25"`, see [^1]. What is a source of your example?

[^1]: https://github.com/google/benchmark/blob/main/docs/user_guide.md#output-formats

Reply:
The result of the local run. Do you have different values locally?

Reply:
Updated:

```diff
diff --git a/perf/lua/benchmark.lua b/perf/lua/benchmark.lua
index 7c04ae0867..e76d7bdb6a 100644
--- a/perf/lua/benchmark.lua
+++ b/perf/lua/benchmark.lua
@@ -35,6 +35,7 @@ local json = require('json')
 local fio = require('fio')
 local argparse = require('internal.argparse')
 local tarantool = require('tarantool')
+local datetime = require('datetime')
 
 local M = {}
 
@@ -134,11 +135,23 @@ local function load_average()
     return { string.match(loadavg_buf, loadavg_re) }
 end
 
+-- Google Benchmark reports contains a date in ISO 8061 format,
+-- example: 2013-07-01T17:55:13-07:00.
+-- See an implementation of `LocalDateTimeString()` in
+-- Google Benchmark library source code [1].
+-- The function return a ISO 8061 formatted timestamp.
+--
+-- 1. https://github.com/google/benchmark/blob/65668db27365d29ca4890cb2102e81acb6585b43/src/timers.cc#L209
+-- @return e.g. 2024-07-17T12:46:31+03:00
+local function iso_8061_timestamp()
+    return datetime.new(os.date('*t')):format('%FT%T.%f%z')
+end
+
 local function perf_context()
     return {
         build_flags = tarantool.build.flags,
         build_target = tarantool.build.target,
-        date = os.date('%Y-%m-%d %H:%M:%S'),
+        date = iso_8061_timestamp(),
         host_name = io.popen('hostname'):read(),
         load_avg = load_average(),
         tarantool_version = tarantool.version,
```

Reply:
Thanks for the fix!

LGTM, after fixing the final nit:
In the created `datetime` object, the timezone isn't specified, so it returns `+0000` instead of `+03:00`. Also, the `:` delimiter is missed.

Reply:
> In the created datetime object, the timezone isn't specified, so it returns +0000 instead of +03:00

For compatibility with Google Benchmark reports and for simplicity, I would not set TZ:

```diff
@@ -144,7 +160,8 @@ end
 -- 1. https://github.com/google/benchmark/blob/65668db27365d29ca4890cb2102e81acb6585b43/src/timers.cc#L209
 -- @return e.g. 2024-07-17T12:46:31+03:00
 local function iso_8061_timestamp()
-    return datetime.new(os.date('*t')):format('%FT%T.%f%z')
+    local dt = os.date('*t')
+    return datetime.new(dt):format('%Y-%m-%dT%H:%M:%S') .. '+00:00'
 end
 
 local function perf_context()
```

```lua
tarantool> dt = os.date('*t')
---
...

tarantool> datetime.new(dt):format('%Y-%m-%dT%H:%M:%S') .. '+00:00'
---
- 2024-07-17T19:11:29+00:00
...
```

> Also, the : delimiter is missed.

Wikipedia says that both variants are ok [^1]:

> The [UTC offset](https://en.wikipedia.org/wiki/UTC_offset) is appended directly to the time instead of "Z" suffix above; other [nautical time](https://en.wikipedia.org/wiki/Nautical_time) zone letters are not used. The offset is applied to [UTC](https://en.wikipedia.org/wiki/UTC) to get the [civil time](https://en.wikipedia.org/wiki/Civil_time) in the designated [time zone](https://en.wikipedia.org/wiki/Time_zone) in the format '±[hh]:[mm]', '±[hh][mm]', or '±[hh]'.

Also see "Date and time — Representations for information interchange Part 1: Basic rules" (BS ISO 8601-1:2019) p. 28:

```
4.3.13 Timeshift
A timeshift, often used in the representation of local standard time against UTC, is represented as follows:

a) Basic, hours and minutes: [±][hour][min] or [“Z”]
EXAMPLE 1 ‘+0500’ or ‘Z’
b) Basic, hoursonly: [±][hour] or [“Z”]
EXAMPLE 2 ‘+05’ or ‘Z’
c) Extended, hours and minutes: [±][hour][“:’][min] or [“Z”]
EXAMPLE 3 ‘+05:00’ or ‘Z’

The UTC designator ["Z"] indicates that there is no time shift from UTC of day and is functionally equivalent to the expressions ‘+0000’ and ‘+00:00’. The time shift shall be expressed as positive(i.e. with the leading plus sign [“+”]) if it is ahead of or equal to UTC, and as negative(i.e. with the leading minus sign[“-”]) if it is behind UTC.
```

[^1]: https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators